ruby/bps-kafka/lib/bps/kafka.rb

```diff
@@ -0,0 +1,42 @@
+require 'bps'
+require 'kafka'
+
+module BPS
+  module Kafka
+    # ReliablePublisher is a slow, but reliable Kafka Publisher.
+    # There is no batching at all, messages are emitted synchronously.
+    class ReliablePublisher < BPS::Publisher::Abstract
+      DELIVER_MESSAGE_RETRIES = 3 # 1 by kafka-ruby's Kafka::Client default, increase slightly
+
+      class Topic < BPS::Publisher::Topic::Abstract
+        def initialize(client, topic_name)
+          @client = client
+          @topic_name = topic_name
+        end
+
+        def publish(msg_data)
+          @client.deliver_message(msg_data, topic: @topic_name, retries: DELIVER_MESSAGE_RETRIES)
+          nil
+        end
+      end
+
+      def initialize(seed_brokers, client_id: nil)
```

Comment:
a note for future PR(s): we should make retries configurable, i.e.: `, retries: DELIVER_MESSAGE_RETRIES` and store as instance var.